### DISCUSSION TITLE
Storing Null In Cache

### DISCUSSION THREAD

**Discussion by kellerjmrtn** [laravel/framework#52449] [General]
According to [PSR-16](https://www.php-fig.org/psr/psr-16/) (which `Illuminate\Cache\Repository` implements)

> **Cache Misses** - A cache miss will return null and therefore detecting if one stored `null` is not possible. This is the main deviation from PSR-6's assumptions.

If a `null` value is fundamentally indistinguishable from simply not setting the value at all, is there any reason to set a `null` value? For example (using Redis store, for example)

```php
Cache::set('some_null_value', null);
```

```
// In Redis

127.0.0.1:6379[1]> keys *
1) "my_cache_prefix:some_null_value"

127.0.0.1:6379[1]> get "my_cache_prefix:some_null_value"
"N;"
```

Despite the key physically being set in Redis, it's seemingly indistinguishable from a key that was never set:

```php
Cache::get('some_null_value'); // null
Cache::has('some_null_value'); // false

// But also

Cache::get('some_value_that_was_never_set'); // null
Cache::has('some_value_that_was_never_set'); // false
```

Therefore, does it even make sense to set a `null` value in cache in the first place? Admittedly I don't know implementation specifics, so it's possible not doing so could cause some undesired behavior somewhere that I'm unaware of. But it seems like avoiding setting `null` could potentially have a few (minor) performance benefits:
1. Save a bit of storage space in cache (especially if this happens repeatedly)
2. Save an extra write operation on `put`, `remember`, etc

### Suggested Change
Since `Illuminate\Cache\Repository` *does* set a real key/value pair in cache when the value is `null`, I'm suggesting/wondering whether it makes sense to update that implementation to simply not set the key/value pair at all

**Comment by henzeb** [comment]
I'd go with Cache:;get('key')?:null. I completely agree with you...
